src/Node/Node.php

```diff
@@ -82,7 +84,7 @@ public function __toString()
     public function compile(Compiler $compiler)
     {
         foreach ($this->nodes as $node) {
-            $node->compile($compiler);
+            $compiler->subcompile($node);
```

Comment:
We'd better not: this line allows the compiler to accurately tell which node is leaking via "echo". If I remove it, the error/deprecation message will report about the wrong node (the wrapping one).
Can't we keep this line? Why if not? Calling subcompile instead of compile looks OK to me also, semantically speaking.

Reply:
note that `$compiler->subcompile($node);` is already calling `$node->compile($compiler);`